ReactCommon/react/renderer/uimanager/UIManagerBinding.cpp

```diff
@@ -140,6 +140,63 @@ void UIManagerBinding::invalidate() const {
   uiManager_->setDelegate(nullptr);
 }
 
+void measureFromShadowTree(
```

Comment:
I wanted to break this work out into a method. It's my understanding c++ doesn't allow you to introduce functions inside functions, so I had to do it here. Is that right? Is there not a way for this code to show up closer to where it's used?

Reply:
This seems correct. You could maybe move it closer by making a lambda, or a macro, but those introduce their own set of tradeoffs.

You probably want to make this a proper private method of the `UIManagerBinding` class though.